pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerBuilderImpl.java

```diff
@@ -328,7 +328,9 @@ private ProducerBuilderImpl(PulsarClientImpl client, ProducerConfigurationData c
     /**
      * Use this config to automatically create an initial subscription when creating the topic.
      * If this field is not set, the initial subscription will not be created.
-     * This method is limited to internal use
+     * If this field is set but the broker's `allowAutoSubscriptionCreation` is disabled, the producer will fail to
+     * be created.
+     * This method is limited to internal use.
```

Comment:
Nit: it could be helpful to explain to end users when this method is used, since the method itself isn't limited to internal use.